src/blockchain/compact_filters/nakamoto.rs

```diff
@@ -0,0 +1,677 @@
+            .map_err(nakamoto_client::Error::from)
+            .map_err(CbfError::from)?
+            .expect("Block at given height doesn't exist");
+        Ok(header.block_hash())
+    }
+}
+
+impl GetTx for CbfNode {
+    fn get_tx(&self, _txid: &Txid) -> Result<Option<Transaction>, crate::Error> {
+        Ok(None)
+    }
+}
+
+impl GetHeight for CbfNode {
+    fn get_height(&self) -> Result<u32, crate::Error> {
+        Ok(self
+            .client_handle
+            .get_tip()
+            .map_err(nakamoto_client::Error::from)
+            .map_err(CbfError::from)?
+            .0 as u32)
+    }
+}
+
+impl WalletSync for CbfNode {
+    fn wallet_setup<D: BatchDatabase>(
+        &self,
+        database: &mut D,
+        progress_update: Box<dyn crate::blockchain::Progress>,
+    ) -> Result<(), crate::Error> {
+        let db_scripts = database.iter_script_pubkeys(None)?;
+        self.client_handle
+            .watch(db_scripts.iter().cloned())
+            .map_err(nakamoto_client::Error::from)
+            .map_err(CbfError::from)?;
+
+        // Rescan from previous block of last_sync_height, to trigger the event loop.
+        self.scan(self.last_sync_height.get().saturating_sub(1), db_scripts);
+
+        let mut internal_max_deriv = None;
+        let mut external_max_deriv = None;
+
+        // Add all broadcasted transactions to db
+        for tx in self.broadcasted_txs.take().iter() {
+            debug!("Processing braodcasted transaction : {}", tx.txid());
+            add_tx(
+                database,
+                tx,
+                None,
+                None,
+                &mut internal_max_deriv,
+                &mut external_max_deriv,
+            )?;
+        }
+
+        loop {
+            // TODO: Investigate why this is causing error
+            // if self.last_sync_height.get() == self.get_height()? {
+            //     break;
+            // }
+            match self.get_next_event()? {
+                Event::BlockConnected { hash, height, .. } => {
+                    debug!("New Block Found : {} at {}", hash, height);
+                }
+                Event::BlockDisconnected { height, .. } => {
+                    let db_txs = database
+                        .iter_txs(false)?
+                        .iter()
+                        .filter_map(|tx_details| {
+                            if let Some(block_time) = &tx_details.confirmation_time {
+                                if block_time.height == height as u32 {
+                                    Some(tx_details.txid)
+                                } else {
+                                    None
+                                }
+                            } else {
+                                None
+                            }
+                        })
+                        .collect::<HashSet<_>>();
+
+                    for txid in db_txs {
+                        unconfirm_tx(database, &txid)?;
```

Comment:
Note that nakamoto should already be emitting a `TxStatus::Reverted` for each tx in a reverted block, so you shouldn't have to handle it here.